src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1327,14 +1347,24 @@ private static bool IsNonContextualModifier(SyntaxToken nextToken)
         private bool IsPartialType()
         {
             Debug.Assert(this.CurrentToken.ContextualKind == SyntaxKind.PartialKeyword);
-            switch (this.PeekToken(1).Kind)
+            var nextToken = this.PeekToken(1);
+            switch (nextToken.Kind)
             {
                 case SyntaxKind.StructKeyword:
                 case SyntaxKind.ClassKeyword:
                 case SyntaxKind.InterfaceKeyword:
                     return true;
             }
 
+            if (nextToken.ContextualKind == SyntaxKind.RecordKeyword)
```

Comment:
>if (nextToken.ContextualKind == SyntaxKind.RecordKeyword) [](start = 12, length = 57)

How did we deal with partial records before?

Reply:
For type declarations like `partial record C ...`, we were relying on `IsPartialMember` returning `true` (didn't investigate how/why), which resulted in same behavior but arguably for the wrong reason.

---
In reply to: [580451252](https://github.com/dotnet/roslyn/pull/51252#discussion_r580451252) [](ancestors = 580451252)